lib/Doctrine/ODM/PHPCR/Tools/Helper/TranslationConverter.php

```diff
@@ -0,0 +1,256 @@
+
+/**
+ * Migrate a field that has become translated or changed its translation
+ * strategy.
+ *
+ * To avoid problems with large repositories, the converter is operating in a
+ * batched mode. Call it repeatedly and call flush and clear between calls.
+ *
+ * You need to call `save()` on the PHPCR session, rather than `flush()` on
+ * the document maanger. To avoid unexpected results, it is recommended to
+ * `clear()` the document manager before starting to convert and after having
+ * saved the PHPCR session.
+ *
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+class TranslationConverter
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $dm;
+
+    /**
+     * @var int Number of documents to process per batch.
+     */
+    private $batchSize;
+
+    /**
+     * @param DocumentManagerInterface $dm
+     * @param int                      $batchSize
+     */
+    public function __construct(DocumentManagerInterface $dm, $batchSize = 200)
+    {
+        $this->dm = $dm;
+        $this->batchSize = $batchSize;
+    }
+
+    /**
+     * Migrate content into the new translation format and remove the old properties.
+     *
+     * This does not commit the changes to the repository. Call save on the
+     * PHPCR session *after each batch*. Calling flush on the document manager
+     * *is not enough*.
+     *
+     * When un-translating, the current locale and language fallback is used.
+     * When translating, the new properties are copied into all languages.
+     *
+     * To convert all fields into a translation, you can pass an empty array
+     * for $fields and the information is read from the metadata. The fields
+     * are mandatory when converting fields back to non-translated.
+     *
+     * To convert a single field from translated to non-translated, simply
+     * specify that field.
+     *
+     * If you convert an existing translation, you need to specify the name of
+     * the strategy that was previously used. The name is the one you would use
+     * for DocumentManagerInterface::getTranslationStrategy, so "attribute" or
+     * "child".
+     *
+     * The current strategy is read from the document metadata.
+     *
+     * @param string        $class                FQN of the document class
+     * @param array         $fields               List of fields to convert. Required when making a
+     *                                            field not translated anymore
+     * @param string        $previousStrategyName Name of previous strategy or "none" if field was not
+     *                                            previously translated
+     * @param array|boolean $strict               Whether to convert descending classes of $class as well.
+     *                                            Pass an empty array if you want to collect which documents
+     *                                            had to be skipped. Pass `false` to convert descendants.
```

Comment:
what do you mean "for return value"? because its pass by ref? another option could be to store this information in a class property so that no pass by ref is needed ..

BTW in that case one could also choose to move the return value into a class property so that one could return the array of class names from the method .. 

but yeah it feels a bit like choosing a poison